# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/performance/Performance.cpp
# Commit: 3cd097a7b017
# Full Hash: 3cd097a7b017154cf5f7a2d9ed97c7891d3074f4
# Author: Will Hawkins <whawkins@mozilla.com>
# Date: 2019-04-10 09:58:56
# Regressor Bug: 1539006
# File Overlap Count: 1
# Description:
#   Bug 1539006: Properly implement Performance Timeline Level 2 w3c spec. r=baku,flod
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D26654
# ==============================================================================

diff -r ea2baf2c9ae7 -r 3cd097a7b017 dom/performance/Performance.cpp
--- a/dom/performance/Performance.cpp	Tue Apr 09 20:32:54 2019 +0000
+++ b/dom/performance/Performance.cpp	Wed Apr 10 00:14:39 2019 +0000
@@ -596,8 +596,23 @@
   if (mObservers.IsEmpty()) {
     return;
   }
-  NS_OBSERVER_ARRAY_NOTIFY_XPCOM_OBSERVERS(mObservers, PerformanceObserver,
-                                           QueueEntry, (aEntry));
+
+  nsTObserverArray<PerformanceObserver*> interestedObservers;
+  nsTObserverArray<PerformanceObserver*>::ForwardIterator observerIt(
+      mObservers);
+  while (observerIt.HasMore()) {
+    PerformanceObserver* observer = observerIt.GetNext();
+    if (observer->ObservesTypeOfEntry(aEntry)) {
+      interestedObservers.AppendElement(observer);
+    }
+  }
+
+  if (interestedObservers.IsEmpty()) {
+    return;
+  }
+
+  NS_OBSERVER_ARRAY_NOTIFY_XPCOM_OBSERVERS(
+      interestedObservers, PerformanceObserver, QueueEntry, (aEntry));
 
   if (!mPendingNotificationObserversTask) {
     RunNotificationObserversTask();